page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java

```diff
@@ -50,93 +87,46 @@ public Page getPage(Class<? extends Page> page) throws PageInitializationExcepti
-            } else {
-                try {
-                    URL currentUrl = new URL(PageFactory.getWebDriver().getCurrentUrl());
-                    String finalUrl = new URL(currentUrl.getProtocol(), currentUrl.getHost(), currentUrl.getPort(),
-                            ((PageEntry) annotation).url()).toString();
-                    PageFactory.getWebDriver().navigate().to(finalUrl);
-                } catch (MalformedURLException ex) {
-                    LOG.error("Failed to get current url", ex);
-                }
+    private static Page bootstrapPage(Class<?> page, WebDriver driver) throws PageInitializationException {
+        if (page != null) {
+            try {
+                @SuppressWarnings("unchecked")
+                Constructor<Page> constructor = ((Constructor<Page>) page.getConstructor(WebDriver.class));
+                constructor.setAccessible(true);
+                Page currentPage = constructor.newInstance(driver);
+                PageContext.setCurrentPage(currentPage);
+                return currentPage;
+            } catch (NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+                throw new PageInitializationException("Failed to initialize page '" + page + "'", e);
             }
-
-            return bootstrapPage(pageClass);
-        }
-
-        throw new AutotestError("WebElementsPage " + title + " doesn't have fast URL in PageEntry");
-    }
-
-    /**
-     * Redirect to WebElementsPage by WebElementsPage Entry url value
-     *
-     * @param title a {@link java.lang.String} object.
-     * @return a WebElementsPage object.
-     * @throws PageInitializationException TODO
-     */
-    public static Page changeUrlByTitle(String title) throws PageInitializationException {
-        if (null != PageContext.getCurrentPage()) {
-            PageContext.setCurrentPage(changeUrlByTitle(pagesPackage, title));
-        }
-        if (null == PageContext.getCurrentPage()) {
-            PageContext.setCurrentPage(changeUrlByTitle(pagesPackage, title));
-        }
-        if (null == PageContext.getCurrentPage()) {
-            throw new AutotestError("WebElementsPage Object with title " + title + " is not registered");
         }
-        return PageContext.getCurrentPage();
-    }
-
-    /**
-     * <p>
-     * Get WebElementsPage by PageEntry title </p>
-     *
-     * @param packageName a {@link java.lang.String} object.
-     * @param title a {@link java.lang.String} object.
-     * @return a WebElementsPage object.
-     * @throws PageInitializationException {@inheritDoc}
-     */
-    public Page getPage(String packageName, String title) throws PageInitializationException {
-        return bootstrapPage(getPageClass(packageName, title));
+        return null;
     }
 
     /**
-     *
      * @param packageName TODO
      * @param title TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)